pygmt/accessors.py

```diff
@@ -131,33 +144,35 @@ def __init__(self, xarray_obj):
     @property
     def registration(self):
         """
-        Registration type of the grid, either 0 (Gridline) or 1 (Pixel).
+        Grid registration type :class:`pygmt.enums.GridRegistration`.
         """
         return self._registration
 
     @registration.setter
     def registration(self, value):
-        if value not in {0, 1}:
+        # TODO(Python>=3.12): Simplify to `if value not in GridRegistration`.
+        if value not in GridRegistration.__members__.values():
             msg = (
-                f"Invalid grid registration value: {value}, should be either "
-                "0 for Gridline registration or 1 for Pixel registration."
+                f"Invalid grid registration: '{value}'. "
+                "Should be either GridRegistration.GRIDLINE or GridRegistration.PIXEL."
```

Comment:
Should we still mention the integer value of the enum? E.g. have the error message be:

```
                "Should be either GridRegistration.GRIDLINE (0) or GridRegistration.PIXEL (1)."
```

Thoughts?

Reply:
Sounds good. Applied in d9bfa3350d8c9ab0dcaea9f56a0077f26264afeb and 3c0d77b46e57e875db2cc06d7c18e83cf2abf857.